### DISCUSSION TITLE
./runtests failed due to python version

### DISCUSSION THREAD

**Discussion by bpatto10** [idaholab/moose#19942] [Q&A Getting Started] [answered]
I built moose from scratch using the web instructions. Everything seemed fine, until I tried the  runtests:

/opt/moose/test$ ./run_tests -j4 -v --dry-run
"TestHarness" requires python version 3.6 or greater, version 2.7 is being used.
 /opt/moose/test$ python -V
Python 3.9.7

I used an alias in my .bashrc file to ensure that python used version 3.9.7 during the various build processes. Is it possible that one of the configure scripts is using 2.7 in spite of my setting it via .bashrc?

**Comment by GiudGiud** [comment, marked as answer]
Hello

How did you set it in your bashrc ? Just with an alias?

Guillaume

**Reply by bpatto10** [reply]
Yes, I just used an alias. I may need to make a tmp link for /usr/bin/python --> python3. I was trying to avoid doing this step as many older linux packages use 2.7.   I forgot to mention that I am running Ubuntu 21.04.

**Reply by milljm** [reply]
There is no Linux 'standard' for setting a python default. So we've had to leave the shebang as:

```bash
#!/usr/bin/env python
```
For some Linux distros, that is still Python 2.x, while others it is 3.x.

I've argued making the shebang `#!/usr/bin/env python3`, but I think I got voted out. If an alias does not work, you can try to alter the shebang in `moose/test/run_tests` to use my suggestion... If you do, please let us know if that worked.
